cluster/gce/upgrade.sh

```diff
@@ -439,6 +439,17 @@ function do-node-upgrade() {
   echo "== Finished upgrading nodes to ${KUBE_VERSION}. ==" >&2
 }
 
+function update-coredns-config() {
+  # Deletes the existing CoreDNS ConfigMap so that the latest can be applied.
+  echo "== Deleting existing CoreDNS ConfigMap. ==" >&2
+  "${KUBE_ROOT}/cluster/kubectl.sh" delete configmap coredns --namespace=kube-system 2>/dev/null
+
+  echo "== Waiting for Addon Manager to restore CoreDNS ConfigMap. ==" >&2
```

Comment:
Just sanity check --- do we have any upgrade suite that runs with kube-dns, in which case addon manager might never restore this?

Reply:
good point.  The kube-dns scenario slipped my mind.  We should only call `update-coredns-config` if coredns is set as the cluster dns.  This is stored in an env variable during upgrade (i recall/hope).